Kernel/Arch/x86/ASM_wrapper.h

```diff
@@ -6,6 +6,11 @@
 
 #pragma once
 
+#include <AK/Platform.h>
+#if !ARCH(I386) && !ARCH(X86_64)
+#   error "Trying to include x86 only header on non x86 platform"
```

Comment:
I think we could re-write this as a static_assert, and then put it behind a macro defined in `Platform.h` that all of these files could share? 

Seems like that would be a nice improvement over copying the same logic to all the headers?

Reply:
That's a good idea

Reply:
Do you mean something like:

```
#include <AK/platform.h>
ENSURE_ARCH(x86);
```

Reply:
Yup!

Reply:
OK, done